### PR TITLE
task 8 get/api/reviews

### PR DESCRIPTION
**Pull request by sasjaWijdenes**
added full functionality and TTD

### CODE REVIEW

**Review by corkesw — COMMENTED** [review]
The solution for the first part of the ticket looks good. We are also asked to do the following:

Queries
The end point should also accept the following query:
- topic, which filters the articles by the topic value specified in the query. If the query is omitted the endpoint should respond with all articles.

Once this is completed and tested, please submit another PR.

Please note that you have committed the test data here which should hot have been included. Be careful about using `git add .`

File: db/models/reviews.models.js [hunk 1]
```diff
@@ -1,5 +1,24 @@
 const db = require("../connection.js");
 
+exports.fetchAllReviews = (sort) => {
+  const queryValues = [],
+    categories = ["euro game", "dexterity", "social deduction", undefined];
+  let queryString = `SELECT reviews.*, COUNT(comments.review_id)::INT AS comment_count FROM reviews LEFT JOIN comments ON reviews.review_id = comments.review_id `;
+  if (!categories.includes(sort))
```
Comment (two-triangles) [inline db/models/reviews.models.js:7]: Is there a test for this behaviour?

File: db/models/reviews.models.js [hunk 2]
```diff
@@ -1,5 +1,24 @@
 const db = require("../connection.js");
 
+exports.fetchAllReviews = (sort) => {
+  const queryValues = [],
+    categories = ["euro game", "dexterity", "social deduction", undefined];
```
Comment (two-triangles) [inline db/models/reviews.models.js:5]: this is currently an undeclared variable - please assign this with const

File: db/models/reviews.models.js [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+  let queryString = `SELECT reviews.*, COUNT(comments.review_id)::INT AS comment_count FROM reviews LEFT JOIN comments ON reviews.review_id = comments.review_id `;
+  if (!categories.includes(sort))
+    return Promise.reject({ status: 404, msg: "No such category" });
+  if (sort) {
```
Comment (two-triangles) [inline db/models/reviews.models.js:9]: I think this variable should be called category as we have named the greenlist of possible values for this variable, "categories"

File: db/models/reviews.models.js [hunk 4]
```diff
@@ -1,5 +1,24 @@
...
+    queryString += `WHERE reviews.category = $1 `;
+    queryValues.push(sort);
+  }
+  queryString += `GROUP BY reviews.review_id;`;
+  return db
+    .query(queryString, queryValues)
+    .then(({ rows: reviews }) => {
+      return reviews;
+    })
+    .catch((err) => console.log(err));
```
Comment (two-triangles) [inline db/models/reviews.models.js:19]: There should not be a catch block in your model. The catch block in your controller will catch and pass it on to your error handling middleware

**Pull request merged**